packages/lesswrong/lib/collections/posts/helpers.ts

```diff
@@ -32,6 +32,46 @@ export const postGetLinkTarget = function (post: PostsBase|DbPost): string {
   return !!post.url ? '_blank' : '';
 };
 
+export const BOOKUI_LINKPOST_WORDCOUNT_THRESHOLD = 800;
+
+interface LinkpostDetectionResult {
+  isLinkpost: boolean;
+  linkpostDomain?: string;
+}
+
+const parseUrlHostname = (url: string): string | undefined => {
+  try {
+    const urlWithProtocol = url.slice(0, 4) === 'http' ? url : `https://${url}`;
+    return new URL(urlWithProtocol).hostname;
+  } catch {
+    return undefined;
+  }
+};
+
+// Detect if a post is a linkpost and get the domain
+export const detectLinkpost = (
+  post: { url?: string | null; contents?: { wordCount?: number | null } | null },
+  rssFeedDomain?: string | null
+): LinkpostDetectionResult => {
+  if (!post.url) {
+    return { isLinkpost: false };
+  }
+
+  const linkpostDomain = parseUrlHostname(post.url);
+  const wordCount = post.contents?.wordCount ?? 0;
+  
+  if (wordCount < BOOKUI_LINKPOST_WORDCOUNT_THRESHOLD) {
+    return { isLinkpost: false, linkpostDomain };
+  }
```

Comment:
I don't think this makes sense to have as part of `detectLinkpost`.  In `[LWPostsPageHeader.tsx](https://github.com/ForumMagnum/ForumMagnum/pull/11134/files#diff-703b6d2907452813b0c40c3a53bbf80abefe3575fa69275ff4657071b2c633b1)` this condition was part of the display logic, but I think it should just stay there.